pkg/crds/names.go

```diff
@@ -0,0 +1,300 @@
+package crds
+
+import "github.com/rancher/rancher/pkg/features"
+
+// RequiredCRDs returns a list of CRD to install based on enabled features.
+func RequiredCRDs() []string {
+	requiredCRDS := BasicCRDs()
+	if features.ProvisioningV2.Enabled() {
+		requiredCRDS = append(requiredCRDS, ProvisioningV2CRDs()...)
+		if features.RKE2.Enabled() {
+			requiredCRDS = append(requiredCRDS, RKE2CRDs()...)
+		}
+		if features.EmbeddedClusterAPI.Enabled() {
+			requiredCRDS = append(requiredCRDS, CAPICRDs()...)
+		}
+		if features.Fleet.Enabled() {
+			requiredCRDS = append(requiredCRDS, "managedcharts.management.cattle.io")
+		}
+	}
+	if features.Fleet.Enabled() {
+		requiredCRDS = append(requiredCRDS, FleetCRDs()...)
+	}
+	if features.MCM.Enabled() {
+		requiredCRDS = append(requiredCRDS, MCMCRDs()...)
+	}
+	if features.Auth.Enabled() {
+		requiredCRDS = append(requiredCRDS, AuthCRDs()...)
+	}
+	if features.MonitoringV1.Enabled() {
+		requiredCRDS = append(requiredCRDS, MonitoringV1CRDs()...)
+	}
+	return requiredCRDS
+}
+
+// BasicCRDs returns a list of CRD names needed to run rancher.
+func BasicCRDs() []string {
+	return []string{
+		"apps.catalog.cattle.io",
+		"clusterrepos.catalog.cattle.io",
+		"operations.catalog.cattle.io",
+		"apiservices.management.cattle.io",
+		"clusters.management.cattle.io",
+		"clusterregistrationtokens.management.cattle.io",
+		"features.management.cattle.io",
+		"podsecurityadmissionconfigurationtemplates.management.cattle.io",
+		"preferences.management.cattle.io",
+		"settings.management.cattle.io",
+		"navlinks.ui.cattle.io",
+	}
+}
+
+// ProvisioningV2CRDs returns a list of CRD names needed for ProvisioningV2.
+func ProvisioningV2CRDs() []string {
```

Comment:
> Another option would be always to install all CRDs.

I don't think we can do this, would be a breaking change from how we operate today and may have other consequences.

My basic idea here would be to use a tag like `//+rancher:featureV2Prov` to indicate which of these functions a CRD belongs to, but that's going to be a non-trivial amount of extra code to produce. We can discuss offline and open a separate issue if we want to take this approach, this could increase the scope of the PR by a lot which I don't like.